db/hints/internal/hint_storage.cc

```diff
@@ -50,16 +50,24 @@ using hints_ep_segments_map = std::unordered_map<unsigned, std::list<fs::path>>;
 // map: IP -> map: shard -> segments
 using hints_segments_map = std::unordered_map<sstring, hints_ep_segments_map>;
 
-future<> scan_for_hints_dirs(const fs::path& hints_directory, std::function<future<> (fs::path dir, directory_entry de, unsigned shard_id)> f) {
```

Comment:
An advantage of using std::function here is that you can give names to its arguments. This is lost when using concepts.

Reply:
This can be "simulated" using comments:
```cc
std::invocable<Func, fs::path /* hint_directory */, directory_entry, unsigned /* shard_id */>
```